### DISCUSSION TITLE
Connecting a Windows Virtual Machine with the Camera

### DISCUSSION THREAD

**Discussion by elias-wahl** [linux-surface/linux-surface#1374] [Q&A] [answered]
Hey :) I have a Windows Surface 7 Pro and yesterday I put the latest Linux Mint (21.3) and the linux-surface-kernel on there (without dual boot). Then I used Qemu to build a Win10 virtual machine. I know that the linux-surface-kernel does not support the camera on the Surface Pro 7. However, I wondered: is there an option to connect the camera to the Windows running in the virtual machine?
Thank you guys in advance!

**Comment by dev-nis** [comment]
As far as I know, this is not possible. Because if the host already has no clue how to handle the camera, then it can't pass the camera to the VM. For example, if cam lists no cameras, then the host doesn't even know there is one, so how could it tell the VM to use it? 

(Please correct me if I'm wrong)

**Reply by StollD** [reply]
This is done by creating a virtual device inside the VM and then letting the VM and the Linux kernel forward all traffic to that device to the real one.

That means that you can run the real Windows driver in the VM, with the real hardware on the host, even if the Linux kernel itself has no idea what the device is doing.

**Reply by dev-nis** [reply]
Oh, interesting! I didn't know that. Thank's for explaining!

**Comment by StollD** [comment, marked as answer]
This probably won't work. 

Even if you managed to pass through the PCI device of the IPU, and the I2C (?) devices of the sensors, you would still need to find a way to connect IPU and sensors. That information is stored in an ACPI table, and while it is possible to boot VM's with custom ACPI tables, it might rely too much on the actual hardware to successfully virtualize it.
